bool is_auto
uint32 percentage